commcare_connect/templates/opportunity/add_visits_existing_users.html

```diff
@@ -0,0 +1,117 @@
+{% extends "base.html" %}
+{% load static %}
+{% load crispy_forms_tags %}
+
+{% block title %}{{ request.org }} - {{ opportunity.name }}{% endblock %}
+
+{% block content %}
+  <div class="container">
+    <h1 class="display-5">Edit Opportunity</h1>
+    <h4>Adding Visits for Existing Users</h4>
+    <form method="post"
+          x-data="{
+            additionalVisits: 0,
+            selected_users: [],
+            isSelected (id) { return this.selected_users.find(el => el === id)},
+            totalVisits() { return this.selected_users.length * this.additionalVisits },
+            calculatedBudget() { return this.totalVisits() * '{{ budget_per_visit }}' }
+          }"
+    >
+      {% csrf_token %}
+      {% if form.selected_users.errors %}
+        <div class="alert alert-danger alert-dismissible fade show" role="alert">
+          Please select users to update.
+          <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
+        </div>
+      {% endif %}
+      {% if opportunity_claims %}
+        <table class="table table-custom border">
+          <thead class="table-secondary">
+          <tr>
+            <th></th>
+            <th>Name</th>
+            <th>Max Visits</th>
+            <th>Used Visits</th>
+            <th>End date</th>
+          </tr>
+          </thead>
+          <tbody>
+          {% for opp_claim in opportunity_claims %}
+            <tr>
+              <td>
+                <input type="checkbox" name="selected_users" id="id_selected_users" value="{{ opp_claim.id }}"
+                       x-model="selected_users"
+                       class="form-check-input {% if form.selected_users.errors %}is-invalid{% endif %}"/>
+              </td>
+              <td>{{ opp_claim.opportunity_access.user.username }}</td>
+              <td>{{ opp_claim.max_payments }}</td>
+              <td>{{ opp_claim.opportunity_access.visit_count }}</td>
+              <td>{{ opp_claim.end_date }}</td>
+            </tr>
+          {% endfor %}
+          </tbody>
+        </table>
+        <div class="row mb-2 py-2">
+          <div class="col-sm">
+            <label for="id_additional_visits" class="form-label requiredField">
+              Additional Visits per user
+            </label>
+            <input type="number"
+                   class="numberinput form-control {% if form.additional_visits.errors %}is-invalid{% endif %}"
+                   name="additional_visits" id="id_additional_visits"
+                   x-model="additionalVisits"/>
+            {% for error in form.additional_visits.errors %}
+              <div class="invalid-feedback">
+                <strong>{{ error|escape }}</strong>
+              </div>
+            {% endfor %}
+          </div>
```

Comment:
Is there a reason to include the form markup directly instead of using crispy tags?

Crispy does have a field render tag: `{{ form.additional_visits|as_crispy_field }}`

https://django-crispy-forms.readthedocs.io/en/latest/api_templatetags.html#templatetags.crispy_forms_filters.as_crispy_field

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/116/commits/c06cebbdaa8351f00d43d9896b94026f7d93e2c3